Adding spring-cloud-data-flow chart to incubator

trisberg opened the pull request.

trisberg committed:
Adding spring-cloud-data-flow chart to incubator

trisberg commented:
CLA should be signed now

dhilipkumars commented:
/ok-to-test

unguiculus commented on incubator/spring-cloud-data-flow/templates/_helpers.tpl:
```diff
@@ -0,0 +1,34 @@
+{{/* vim: set filetype=mustache: */}}
```
Our best practices suggest to use `fullname` for resource names and `name` for `app` labels. You use `shortname`, which is an adapted version of `fullname`, for both. I think it is fine to abbreviate the name as you do but still stick to our best practices. I'd suggest you create two templates `scdf.fullname` and `scdf.name`.

unguiculus reviewed (changes requested):
Thanks for the PR.

trisberg committed:
Updating chart based on suggested changes

- adding "scdf" prefix to defined templates

- switching to use "data-flow" as the shorter app name for resources

- updating resource labels to be consistent

- using template "name" for labels and "scdf.fullname" for resources

- updated service selectors to match new labels

trisberg commented:
@unguiculus I made some changes trying to follow your suggestions. Not sure about the CI error, any idea what that means? Thanks for taking a look at this.

unguiculus commented on incubator/spring-cloud-data-flow/Chart.yaml:
```diff
@@ -0,0 +1,14 @@
+- https://github.com/spring-cloud/spring-cloud-dataflow-metrics-collector
+icon: https://raw.githubusercontent.com/spring-cloud/spring-cloud-dataflow/master/spring-cloud-dataflow-docs/src/main/docbook/images/logo.png
+maintainers:
+- name: Thomas Risberg
```
Use Github username.

unguiculus commented on incubator/spring-cloud-data-flow/templates/metrics-deployment-rabbit.yaml:
```diff
@@ -0,0 +1,60 @@
+  name: {{ template "scdf.fullname" . }}-metrics
+  labels:
+    app: {{ template "name" . }}
+    component: {{ template "name" . }}-metrics
```
Remove the `{{ template "name" . }}` prefix from components.

unguiculus commented on incubator/spring-cloud-data-flow/templates/metrics-deployment-rabbit.yaml:
```diff
@@ -0,0 +1,60 @@
+        imagePullPolicy: {{ .Values.metrics.imagePullPolicy }}
+        ports:
+        - containerPort: {{ .Values.metrics.service.internalPort }}
+        resources:
```
Prefer this:

`{{ toYaml .Values.metrics.resources | indent 10 }}`

unguiculus commented on incubator/spring-cloud-data-flow/templates/NOTES.txt:
```diff
@@ -0,0 +1,15 @@
+  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "scdf.fullname" . }}-server -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  echo http://$SERVICE_IP:{{ .Values.server.service.externalPort }}
+{{- else if contains "ClusterIP"  .Values.service.type }}
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "scdf.fullname" . }}-server" -o jsonpath="{.items[0].metadata.name}")
```
`-l "app={{ template "scdf.fullname" . }}-server"` -> `-l "app={{ template "name" . }},release={{ .Release.Name }},component=server"`

This considers the component change suggested below.

unguiculus commented on incubator/spring-cloud-data-flow/values.yaml:
```diff
@@ -0,0 +1,48 @@
+    type: LoadBalancer
+    externalPort: 80
+    internalPort: 80
+  resources:
```
Don't specify defaults for resources. We've started recommending to leave this a conscious choice of the user. It should also be possible to test charts on systems with little resources, such as Minikube.

Do it like so:

```yaml
  resources: {}
  #  limits:
  #    cpu: 1.0
  #    memory: 2048Mi
  #  requests:
  #    cpu: 0.5
  #    memory: 1024Mi
```

unguiculus commented on incubator/spring-cloud-data-flow/templates/server-config-rabbit.yaml:
```diff
@@ -0,0 +1,40 @@
+            authentication:
+              file:
+                enabled: true
+                users:
```
Credentials should be stored in secrets.

unguiculus replied:
Can you address the secrets, please?

trisberg replied:
I thought I did - I added a `templates/secrets.yaml` file and use Spring Boot property reference place holders instead of the template Values placeholders for the passwords

unguiculus replied:
OK, I'm seeing it now. Please give me some time to fully review the chart.

unguiculus reviewed (changes requested):
It is a best practice to optionally allow passwords to be generated. Many charts do this. Here's an example:

https://github.com/kubernetes/charts/blob/master/stable/mariadb/templates/secrets.yaml

trisberg committed:
Addressing review comments

- adding secrets for data-flow passwords

trisberg commented:
@unguiculus I made the requested changes, anything else I have to do to move this forward, not familiar with your workflow. BTW, the workflow link from the README page results in a 404.

unguiculus commented:
/retest

trisberg commented:
@unguiculus thanks for your help and suggestions for improvements
